### PR TITLE
[API] Tax endpoint

### PR DESCRIPTION
**Pull request by claudiosanches**
Don't merge yet!

TODO:
- [x] GET /taxes
- [x] GET /taxes/&lt;id&gt;
- [x] POST /taxes
- [x] PUT /taxes/&lt;id&gt;
- [x] DELETE /taxes/&lt;id&gt;
- [x] GET /taxes/classes
- [x] POST /taxes/classes
- [x] DELETE /taxes/classes/&lt;slug&gt;
- [x] GET /taxes/count
- [x] GET /taxes/classes/count
- [x] POST/PUT /taxes/bulk

### CODE REVIEW

**Comment by claudiosmweb** [comment]
cc @gedex @maxrice :)

File: includes/api/class-wc-api-taxes.php [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+		try {
+			$id = absint( $id );
+
+			// Permissions check
+			if ( ! current_user_can( 'manage_woocommerce' ) ) {
+				throw new WC_API_Exception( 'woocommerce_api_user_cannot_read_tax_rate', __( 'You do not have permission to read tax rate', 'woocommerce' ), 401 );
+			}
+
+			// Get tax rate details
+			$tax = $wpdb->get_row( $wpdb->prepare( "
```
Comment (maxrice) [inline includes/api/class-wc-api-taxes.php]: might be worth abstracting the retrieval into it's own global function so it's usable by non-API code as well

**Comment by maxrice** [comment]
:+1: LGTM

**Comment by gedex** [comment]
Great work @claudiosmweb ! LGTM too.

File: includes/api/class-wc-api-taxes.php (hunk 1 above)
Comment (claudiosmweb) [inline includes/api/class-wc-api-taxes.php]: Good idea :)
I'll add it on the `WC_Tax` class.

**Comment by claudiosmweb** [comment]
Done :)

**Pull request merged**